internal/backend/b2/b2.go

```diff
@@ -258,7 +261,22 @@ func (be *b2Backend) Remove(ctx context.Context, h restic.Handle) error {
 	defer be.sem.ReleaseToken()
 
 	obj := be.bucket.Object(be.Filename(h))
-	return errors.Wrap(obj.Delete(ctx), "Delete")
+
+	if be.canDelete {
+		err := errors.Wrap(obj.Delete(ctx), "Delete")
+		if err != nil {
+			code, _ := base.Code(err)
```

Comment:
This can't work as the error returned by `obj.Delete` is wrapped.

Reply:
Uups. I thought I did test out the error path but it's been so long I don't even remember.